src/components/PetGrid.tsx

```diff
@@ -1,27 +1,34 @@
 import { useEffect, useState } from 'react';
 import { PetCard } from '@/components/PetCard';
 import { getPets } from '@/data/pets/getPets';
+import { Loader } from '@/components/Loader';
 
 type PetGridProps = {
   refetch: boolean;
 };
 
 export const PetGrid: React.FC<PetGridProps> = ({ refetch }) => {
   const [pets, setPets] = useState<PetData[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
     const fetchPets = async () => {
+      setIsLoading(true);
       try {
         const petsData = await getPets();
         setPets(petsData);
       } catch (error) {
         console.error(error);
+      } finally {
+        setIsLoading(false);
       }
     };
     fetchPets();
   }, [refetch]);
 
-  return (
+  return isLoading ? (
+    <Loader />
+  ) : (
```

Comment:
🟡 What do you think about displaying an error message when an error occurs? Currently, we only have the PetGrid or Loader components, which could lead to an infinite loading state if data can't be fetched. To improve user experience, we could implement something like this:
```
const [errorMessage, setErrorMessage] = useState<string | null>(null);

  useEffect(() => {
    const fetchPets = async () => {
      setIsLoading(true);
      try {
        const petsData = await getPets();
        setPets(petsData);
      } catch (error) {
        throw error
        setErrorMessage('Failed to load pets. Please try again.')
      } finally {
        setIsLoading(false);
      }
    };
    fetchPets();
  }, [refetch]);

if (error) {
  return <div className="text-red-500">{errorMessage}</div>;
}

```
This way, users will see a clear error message instead of being stuck in an infinite loading state.

Reply:
Good idea! I didn't think about that.

Maybe we can create a reusable component (with some nicer design than just a text string) that accepts the error message and displays it to the user?